drivers/spi/spi_context.h

```diff
@@ -227,7 +227,7 @@ void spi_context_buffers_setup(struct spi_context *ctx,
 	if (tx_bufs) {
 		ctx->current_tx = tx_bufs->buffers;
 		ctx->tx_count = tx_bufs->count;
-		ctx->tx_buf = ctx->current_tx->buf;
+		ctx->tx_buf = (const u8_t *)ctx->current_tx->buf;
```

Comment:
You don't actually need the "const" here.  It's already a typecast (that is, we're throwing out typesafety already), and const will be applied by assignment automatically without a cast.

Reply:
An explicit cast is not only usefull for the compiler, but also for the developer. I think is more clear and comprehensive to write the entire cast.
For me is better to write a full explicit cast, instead of a "half" explicit cast (explicit u8_t*, implicit const).